engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java

```diff
@@ -434,6 +434,52 @@ public void testFailingEvaluations() {
     }
   }
 
+  @Test
+  public void testAtomFieldsAreReadable() {
+    var fooFunc =
+        createEnsoMethod(
+            """
+        type My_Type
+            Cons field_1 field_2
+
+        foo x =
+            obj = My_Type.Cons 1 2
+            obj
+        """,
+            "foo");
+    try (DebuggerSession session =
+        debugger.startSession(
+            (SuspendedEvent event) -> {
+              switch (event.getSourceSection().getCharacters().toString().strip()) {
+                case "obj" -> {
+                  DebugScope scope = event.getTopStackFrame().getScope();
+                  DebugValue objValue = scope.getDeclaredValue("obj");
+                  assertThat(objValue.isReadable(), is(true));
+                  assertThat(objValue.isInternal(), is(false));
+                  assertThat(objValue.hasReadSideEffects(), is(false));
+
+                  var field1Prop = objValue.getProperty("field_1");
+                  assertThat(field1Prop.isReadable(), is(true));
+                  assertThat(field1Prop.isNumber(), is(true));
+                  assertThat(field1Prop.asInt(), is(1));
+
+                  assertThat(objValue.getProperties().size(), is(2));
+                  for (var prop : objValue.getProperties()) {
+                    assertThat(
```

Comment:
This test fails, which explain why the atom fields are not visible in the debugger.

Reply:
Fixing https://github.com/enso-org/enso/pull/10661/files#r1690196270 will also fix this test. Should be straightforward.

Reply:
Fixed with https://github.com/enso-org/enso/pull/10661/commits/d34c3e721292c2d567f5a958aac36a0fd026c71a